source/core/CEntryPoints.cpp

```diff
@@ -118,6 +118,45 @@ VIREO_EXPORT Int32 EggShell_PokeMemory(TypeManagerRef tm,
     }
 }
 //------------------------------------------------------------
+//! Allocates enough memory to fit a new object of TypeRef
+VIREO_EXPORT EggShellResult EggShell_AllocateData(TypeManagerRef tm, const TypeRef typeRef, void** dataRefLocation)
+{
+    TypeManagerScope scope(tm);
+    if (typeRef == nullptr || !typeRef->IsValid()) {
+        return kEggShellResult_InvalidTypeRef;
+    }
+
+    *dataRefLocation = nullptr;
+    Int32 topSize = typeRef->TopAQSize();
+    void* pData = THREAD_TADM()->Malloc(topSize);
+    NIError error = typeRef->InitData(pData, (TypeRef)nullptr);
+    if (error != kNIError_Success) {
+        return kEggShellResult_UnableToAllocateData;
+    }
+    *dataRefLocation = pData;
+    return kEggShellResult_Success;
+}
+//------------------------------------------------------------
+//! Deallocates data and frees up memory in dataRef described by typeRef
+VIREO_EXPORT EggShellResult EggShell_DeallocateData(TypeManagerRef tm, const TypeRef typeRef, void* dataRef)
+{
+    TypeManagerScope scope(tm);
+    if (typeRef == nullptr || !typeRef->IsValid()) {
+        return kEggShellResult_InvalidTypeRef;
+    }
+
+    if (dataRef == nullptr) {
+        return kEggShellResult_NullDataPointer;
+    }
+
+    NIError error = typeRef->ClearData(dataRef);
```

Comment:
Add a free call to release the dataRef pointer.

Reply:
Yeah, I'll add that. I couldn't find an easy way to enable memory leak checking from JS.